types/p-log/index.d.ts

```diff
@@ -0,0 +1,13 @@
+// Type definitions for p-log 1.0
+// Project: https://github.com/sindresorhus/p-log#readme
+// Definitions by: BendingBender <https://github.com/BendingBender>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export = pLog;
+
+declare const pLog: pLog;
+
+interface pLog {
+    <T>(): (value: T) => Promise<T>;
```

Comment:
If I remove the generic type and replace it with `any`, then I have to explicitly cast/annotate the incoming value in the `.then()` following the `.then(pLog())`. Which I have to do anyway in TypeScript < 2.4 but TypeScript 2.4 is able to infer the correct type here.

I am not sure what you want me to do here.

To explain why I used an interface with a call signature: I had to somehow combine the `catch()` method into the type which is a reserved word. So namespace was not an option. And using a function type and combining it with the interface that only contains `catch()` via union combinator was also not an option because the type parameter on the function type doesn't allow me to create a `const` binding with such a type.